assets/src/admin/site-scan-notice/theme-with-amp-incompatibility.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { AMP_COMPATIBLE_THEMES_URL } from 'amp-site-scan-notice'; // From WP inline script.
+
+/**
+ * WordPress dependencies
+ */
+import { createInterpolateElement, useContext } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { Themes } from '../../components/themes-context-provider';
+import { isExternalUrl } from '../../common/helpers/is-external-url';
+
+/**
+ * Render a message saying a theme is not AMP compatible.
+ *
+ * @param {Object} props                             Component props.
+ * @param {Object} props.themeWithAmpIncompatibility Theme with AMP incompatibilities.
+ */
+export function ThemeWithAmpIncompatibility( { themeWithAmpIncompatibility } ) {
+	const { fetchingThemes, themes } = useContext( Themes );
+
+	if ( fetchingThemes ) {
+		return null;
+	}
+
+	const themeMeta = themes.find( ( theme ) => theme.stylesheet === themeWithAmpIncompatibility.slug );
+	const themeName = themeMeta?.name?.rendered ?? themeMeta?.name;
+
+	return (
+		<>
+			<p>
+				{ createInterpolateElement(
+					sprintf(
+						// translators: %s stands for a theme name.
+						__( 'AMP compatibility issues discovered with the <b>%s</b> theme.', 'amp' ),
+						themeName,
+					),
+					{
+						b: <strong />,
+					},
+				) }
+			</p>
```

Comment:
There may actually be _two_ themes that have compatibility issues, the parent theme _and_ the child theme. Therefore I suggest that the structure here follow closely what you did in `PluginsWithAmpIncompatibility` and have the list of themes along with the URLs that they have issues detected on.

Reply:
Definitely. Check out https://github.com/ampproject/amp-wp/pull/6859#issuecomment-1059806234 below.